### PR TITLE
Open TODO.html on Mac/Linux using bash

### PR DESCRIPTION
**Pull request by RehanSaeed**
See https://twitter.com/RehanSaeedUK/status/1364326340191789056?s=20

### CODE REVIEW

File: Source/ApiTemplate/.template.config/template.json [hunk 1]
```diff
@@ -694,17 +694,35 @@
   "postActions": [
     {
       "actionId": "3A7C4B45-1F5D-4A30-959A-51B88E82B5D2",
-      "condition": "(!SkipOpenToDo)",
+      "condition": "(!SkipOpenToDo) && (OS == \"Windows_NT\")",
```
Comment (bradygaster) [inline Source/ApiTemplate/.template.config/template.json:697]: So essentially if the customer is on Windows and using PowerShell, is there a degraded experience? Does Power Shell automatically "do something" when you use the templates using Windows, that would be lacking on non-Windows side? Not a complaint - just curious - we *try* to retain fidelity across platforms. Not always possible - your effort to do so is applauded especially on such short and informal notice - but if there's a way I'd love to see it achieved.

Comment (RehanSaeed) [reply Source/ApiTemplate/.template.config/template.json:697]: The template comes with a `TODO.html` file which contains a checklist of improvements you can/should make. We try to open this file in your default browser when you create a new project from the template. You can totally, stop this from happening and skip this step by passing the `--no-open-todo` argument to `dotnet new`.

image

We use a mechanism given to us by the `dotnet new` templating engine to execute arbitrary shell commands to achieve this. This PR, continues to use PowerShell to open `TODO.html` on Windows and adds the ability to use bash on non-Windows operating systems.

**Comment by bradygaster** [comment]
thanks for that info @RehanSaeed. That _seems like_ something that might be defaulted to "off" rather than "on" to be more welcoming to new customers? Granted, I didn't do a `-h` to see what my options were, I just dove in, but I think that most customers might "just dive in," too. I also wonder what the implications to a template having that sort of a post-creation behavior would have by Visual Studio customers. Curious if @sayedihashimi has any thoughts on the idea. Apologies if this has already been discussed and/or debated, I'm a new Boxed user so I'm arguably playing devil's advocate on behalf of new customers for a moment.

**Comment by sayedihashimi** [comment]
I'm not sure if that post action will run when invoked by Visual Studio. If so, it can be conditioned to only run during CLI invocations. Let me know if you need any info on how to accomplish that.

**Comment by RehanSaeed** [comment]
@bradygaster If we removed the option to open the file at the end of project creation altogether, do you think you'd find and open it while looking around the project anyway?

@sayedihashimi I hadn't realized that VS might not support it. Is there an open issue or doc I can look at?

**Comment by sayedihashimi** [comment]
@RehanSaeed if you want a file to be opened in VS after the project is created, you can add the file to the `primaryOutputs`, and it should be opened in VS.

**Comment by RehanSaeed** [comment]
But that will open the file in the VS editor. I'd like to open in the default browser. There is an 'open in browser' post action but it's not yet been implemented (See https://github.com/dotnet/templating/issues/2087).

What is VS's behaviour with the 'Run script' post action? Will it just not work? Will it display a warning/error?

**Comment by sayedihashimi** [comment]
@phenning can you clarify what is the expectation around post actions running in VS?

**Pull request merged**